manager/integration/tests/test_infra.py

```diff
@@ -0,0 +1,69 @@
+import os
+import pytest
+import time
+
+from digitalocean import digitalocean
+from common import get_core_api_client, get_longhorn_api_client
+
+
+def detect_cloudprovider():
+    cloudprovider_name = os.getenv("CLOUDPROVIDER")
+
+    if cloudprovider_name == "digitalocean":
+        cloudprovider = digitalocean()
+        return cloudprovider
+
+
+@pytest.mark.infra
+def test_offline_node():
+    node_worker_label = 'node-role.kubernetes.io/worker'
+    pod_lable_selector = "longhorn-test=test-job"
+
+    k8s_api_client = get_core_api_client()
+    longhorn_api_client = get_longhorn_api_client()
+    cloudprovider = detect_cloudprovider()
+
+    for pod in k8s_api_client.list_namespaced_pod(
+                    'default',
+                    label_selector=pod_lable_selector).items:
+
+        if pod.metadata.name == "longhorn-test":
+            longhorn_test_node_ip = pod.status.host_ip
+
+    for node_item in k8s_api_client.list_node().items:
+        if node_worker_label in node_item.metadata.labels and \
+                node_item.metadata.labels[node_worker_label] == 'true':
+            node_name = node_item.metadata.name
+            node_external_ip = \
```

Comment:
Why do we use IP as node ID? I thought the node name is a natural choice?

Also, this one used an RKE specific label. We may not want that since that limits what setup this test can run.

Reply:
- I rely on node IP to filter digital ocean nodes, and get node's unique ID. as node names can be redundant.

Reply:
We can require the test environment to have different node names. It doesn't make sense to have the same node name in the production environment as well. It will make things easier.